src/theory/sets/theory_sets_private.cpp

```diff
@@ -1234,44 +1234,8 @@ Node mkAnd(const std::vector<TNode>& conjunctions)
   return conjunction;
 } /* mkAnd() */
 
-bool TheorySetsPrivate::propagate(TNode literal)
-{
-  Debug("sets-prop") << " propagate(" << literal << ")" << std::endl;
-
-  // If already in conflict, no more propagation
-  if (d_state.isInConflict())
-  {
-    Debug("sets-prop") << "TheoryUF::propagate(" << literal
-                       << "): already in conflict" << std::endl;
-    return false;
-  }
-
-  // Propagate out
-  bool ok = d_external.d_out->propagate(literal);
-  if (!ok)
-  {
-    d_state.notifyInConflict();
-  }
-
-  return ok;
-} /* TheorySetsPrivate::propagate(TNode) */
-
-OutputChannel* TheorySetsPrivate::getOutputChannel()
-{
-  return d_external.d_out;
-}
-
 Valuation& TheorySetsPrivate::getValuation() { return d_external.d_valuation; }
 
-void TheorySetsPrivate::conflict(TNode a, TNode b)
-{
-  Node conf = explain(a.eqNode(b));
```

Comment:
The new code uses `TheoryInferenceManager::explainConflictEqConstantMerge` which depends on `TheoryInferenceManager::explainConflictEqConstantMerge`. On the other hand, `TheorySetsPrivate::explain` depends on `EqualityEngine::explainEquality`. Although both `explain` the conflict, their code is different. Are they the same?

Reply:
Yes, in the end sets is making custom calls to explain equalities from equality engine, but the code should be the same.  The variants of TheorySetsPrivate::explain that do not call to theory inference manager will be deleted soon.